ci/cloudbuild/builds/generate-libraries.sh

```diff
@@ -32,22 +32,39 @@ bazel run --action_env=GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes \
   --update_ci=false \
   --config_file="${PWD}/generator/integration_tests/golden_config.textproto"
 
-io::log_h2 "Running the generator to update the generated libraries"
-bazel run --action_env=GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes \
-  //generator:google-cloud-cpp-codegen -- \
-  --protobuf_proto_path="${bazel_output_base}"/external/com_google_protobuf/src \
-  --googleapis_proto_path="${bazel_output_base}"/external/com_google_googleapis \
-  --output_path="${PROJECT_ROOT}" \
-  --config_file="${PROJECT_ROOT}/generator/generator_config.textproto"
-
-# TODO(#5821): The generator should run clang-format on its output files itself
-# so we don't need this extra step.
-io::log_h2 "Formatting generated code"
-git ls-files -z -- '*.h' '*.cc' |
-  xargs -P "$(nproc)" -n 1 -0 clang-format -i
-
-io::log_h2 "Updating protobuf lists/deps"
-external/googleapis/update_libraries.sh
+if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
+  io::log_h2 "Running the generator to update the generated libraries"
+  bazel run --action_env=GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes \
+    //generator:google-cloud-cpp-codegen -- \
+    --protobuf_proto_path="${bazel_output_base}"/external/com_google_protobuf/src \
+    --googleapis_proto_path="${bazel_output_base}"/external/com_google_googleapis \
+    --output_path="${PROJECT_ROOT}" \
+    --config_file="${PROJECT_ROOT}/generator/generator_config.textproto"
+else
+  io::log_red "Skipping update of generated libraries."
+fi
+
+if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
+  # TODO(#5821): The generator should run clang-format on its output files itself
+  # so we don't need this extra step.
+  io::log_h2 "Formatting generated code"
+  git ls-files -z -- '*.h' '*.cc' |
+    xargs -P "$(nproc)" -n 1 -0 clang-format -i
+else
+  io::log_red "Only formatting generator golden code."
+  git ls-files -z -- 'generator/integration_tests/golden/internal/*.h' \
+    'generator/integration_tests/golden/*.h' \
+    'generator/integration_tests/golden/internal/*.cc' \
+    'generator/integration_tests/golden/*.cc' |
+    xargs -P "$(nproc)" -n 1 -0 clang-format -i
+fi
+
+if [ -z "${GENERATE_GOLDEN_ONLY}" ]; then
+  io::log_h2 "Updating protobuf lists/deps"
+  external/googleapis/update_libraries.sh
+else
+  io::log_red "Skipping update of generated libraries dependencies."
```

Comment:
"Skipping update of protobuf lists/deps" perhaps, to match non-skipping message.